crates/subspace-networking/src/node.rs

```diff
@@ -424,6 +443,24 @@ impl Node {
             .ok_or(AnnouncePieceError::Announce)
     }
 
+    /// Stop announcing item by its key. Initiate 'stop_providing' Kademlia operation.
+    pub async fn stop_announcing(&self, key: Multihash) -> Result<(), StopAnnouncingError> {
```

Comment:
This is what I was talking about: it would be better if `announce` returned an object that calls internal `stop_announcing` on drop instead of managing the lifecycle manually. We can have `leak` method on it to announce something forever, but the default would be to stop announcing on drop.

Also I I don't remember the libp2p's behavior off top of my head, but if re-announcements are not done automatically, this object would have to do that too to avoid expirations.

Reply:
libp2p contains a way to automatically republish the resource - we need to set `provider_publication_interval` as I did in this PR. 

Self-managing announcement object is one of the possible options, however, depending on the other features like local piece cache, it could have several implementations. I'm not sure yet about all requirements for this feature and I would implement it as a separate task later when we finalize the plotting. I prepared a task for it: https://github.com/subspace/subspace/issues/851

Reply:
The primary idea is not managing republishing, but rather to return an object that on drop will stop publishing, I'm not sure it depends on anything in particular much.

This is similar to how I made subscriptions wrapper here that unsubscribes from the topic once dropped.